### PR TITLE
Integrating SparseQArray in the dynamiqs workflow.

### PR DESCRIPTION
**Pull request by bsaoptima**
In this PR, we are integrating the SparseQArray format to be compatible with `sesolve` & `mesolve` for all types of supported Hamiltonians. We will also implement the test functions to verify the accuracy of our results compared to the classical Dense format.

### CODE REVIEW

File: dynamiqs/time_array.py [hunk 1]
```diff
@@ -30,6 +31,9 @@ def constant(array: ArrayLike) -> ConstantTimeArray:
         _(time-array object of shape (..., n, n) when called)_ Callable object
             returning $O_0$ for any time $t$.
     """
+    if isinstance(array, SparseQArray):
+        return ConstantTimeArray(array)
+
```
Comment (gautierronan) [inline dynamiqs/time_array.py:36]: Here (and below), we should still keep the call to `check_shape` even if the input is a `SparseQArray`. So only the `jnp.asarray` should be discarded here.

**Review by gautierronan — CHANGES_REQUESTED** [review]
That's great! Only a minor comment below, but otherwise it's good with me
